src/main/java/org/opensearch/security/dlic/rest/validation/RequestContentValidator.java

```diff
@@ -393,4 +535,242 @@ public String message() {
+
+    /**
+     * Validates a path-like value (no whitespace allowed)
+     */
+    public static void validatePath(String fieldName, String path, int maxLength) {
+        requireNonEmpty(fieldName, path);
+        validateMaxLength(fieldName, path, maxLength);
+        if (!path.equals(path.trim()) || path.chars().anyMatch(Character::isWhitespace)) {
+            throw new IllegalArgumentException(fieldName + " must not contain whitespace");
+        }
+    }
+
+    /**
+     * Validates a value against an allowed set
+     */
+    public static void validateFieldValueInSet(
+        String fieldName,
+        String value,
+        int maxLength,
+        Set<String> allowedValues,
+        String errorContext
+    ) {
+        requireNonEmpty(fieldName, value);
+        validateMaxLength(fieldName, value, maxLength);
+        if (allowedValues == null || allowedValues.isEmpty()) {
+            throw new IllegalStateException("No allowed " + errorContext + " configured");
+        }
+        if (!allowedValues.contains(value)) {
+            throw new IllegalArgumentException("Invalid " + fieldName + " [" + value + "]. Allowed " + errorContext + ": " + allowedValues);
+        }
+    }
+
+    /**
+     * Validates a JSON object node with non-empty string values
+     */
+    public static void validateObjectWithStringValues(String fieldName, JsonNode node) {
+        if (node == null || node.isNull()) {
+            return;
+        }
+
+        if (!node.isObject()) {
+            throw new IllegalArgumentException(fieldName + " must be an object");
+        }
+
+        if (!node.fieldNames().hasNext()) {
+            throw new IllegalArgumentException(fieldName + " cannot be empty");
+        }
+
+        node.fields().forEachRemaining(entry -> {
+            JsonNode val = entry.getValue();
+            if (!val.isTextual() || val.asText().isEmpty()) {
+                throw new IllegalArgumentException(fieldName + " for key [" + entry.getKey() + "] must be a non-empty string");
+            }
+        });
+    }
+
+    /* ========================================================================
+     * Pre-built Field Validators for Common Use Cases
+     * ======================================================================== */
+
+    /**
+     * Creates a validator for principal values (users, roles, backend_roles, etc.)
+     * @param allowWildcard whether to allow "*" as a standalone value
+     */
+    public static FieldValidator principalValidator(boolean allowWildcard) {
+        return (fieldName, value) -> {
+            if (value instanceof String strValue) {
+                validateSafeValue(fieldName, strValue, MAX_PRINCIPAL_LENGTH, allowWildcard);
+            }
+        };
+    }
+
+    /**
+     * Validator for principal values (no wildcard support)
+     */
+    public static final FieldValidator PRINCIPAL_VALIDATOR = principalValidator(false);
```

Comment:
Let's move these to the applicable subclass of AbstractApiAction

Reply:
no subclasses using this value implement AbstractApiAction, and it should be fine here for now. We already have validatePassword in here that is specific to users API action.